worker/src/email.rs

```diff
@@ -0,0 +1,207 @@
+use futures_util::TryStreamExt;
+use wasm_bindgen::JsCast;
+use wasm_bindgen::JsValue;
+use wasm_bindgen_futures::JsFuture;
+use worker_sys::EmailMessage as EmailMessageExt;
+use worker_sys::ForwardableEmailMessage as ForwardableEmailMessageExt;
+use worker_sys::SendEmail as SendEmailExt;
+
+use crate::ByteStream;
+use crate::EnvBinding;
+use crate::Error;
+use crate::Headers;
+
+pub struct EmailMessage(EmailMessageExt);
+
+unsafe impl Send for EmailMessage {}
+unsafe impl Sync for EmailMessage {}
+
+impl JsCast for EmailMessage {
```

Comment:
I may be wrong, but you may not need all of this boilerplate for types that do not need to implement `EnvBinding`.